src/orange/construct/detail/PostfixLogicBuilder.hh

```diff
@@ -0,0 +1,149 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/construct/detail/PostfixLogicBuilder.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/cont/VariantUtils.hh"
+#include "orange/OrangeTypes.hh"
+
+#include "../CsgTree.hh"
+
+namespace celeritas
+{
+namespace csg
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Recursively construct a logic vector from a node with postfix operation.
+ */
+class PostfixLogicBuilder
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using VecLogic = std::vector<logic_int>;
+    //!@}
+
+    static_assert(std::is_same_v<LocalSurfaceId::size_type, logic_int>,
+                  "unsupported: add enum logic conversion for different-sized "
+                  "face and surface ints");
+
+  public:
+    // Construct with reference to vector to append to
+    explicit inline PostfixLogicBuilder(CsgTree const& tree, VecLogic* logic);
+
+    //! Build from a node ID
+    inline void operator()(NodeId const& n);
+
+    // Visit an actual surface
+    inline void operator()(True const&);
+    inline void operator()(False const&);
+    inline void operator()(Surface const&);
+    inline void operator()(Aliased const&);
+    inline void operator()(Negated const&);
+    inline void operator()(Joined const&);
+
+  private:
+    ContainerVisitor<CsgTree const&, NodeId> visit_node_;
+    VecLogic* logic_;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with reference to the logic expression.
+ */
+PostfixLogicBuilder::PostfixLogicBuilder(CsgTree const& tree, VecLogic* logic)
+    : visit_node_{tree}, logic_{logic}
+{
+    CELER_EXPECT(logic_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Build from a node ID.
+ */
+void PostfixLogicBuilder::operator()(NodeId const& n)
+{
+    return this->visit_node_(*this, n);
```

Comment:
Remove return here and in all other operator() functions

Reply:
Sometimes this is a good thing to do (it ensures that the function you're calling actually returns `void` so you're not discarding the result), but you're right it doesn't add anything here.

Reply:
The confusing part was really the inconsistency: some functions had returns and other didn't